### PR TITLE
[WIP] Fix gradient norm tracking and gradient clipping

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

THIS PR WILL BE SPLIT UP

Reported in Slack

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loops/optimizer/optimizer_loop.py [hunk 1]
```diff
@@ -355,22 +330,3 @@ def _training_step(
...
-
-    def _track_and_norm_grad(self, optimizer: torch.optim.Optimizer) -> Dict[str, float]:
-        """Tracks gradient norms and clips the gradients of all parameters optimized by the current optimizer.
-
-        Args:
-            optimizer: the current optimizer
-        """
-        # track gradient norms
-        grad_norm_dict = {}
-        can_log = (self.trainer.global_step + 1) % self.trainer.log_every_n_steps == 0
```
Comment (carmocca) [inline pytorch_lightning/loops/optimizer/optimizer_loop.py:367]: I don't know why we were checking `can_log`. This is a responsibility of the logger connector.

Perhaps just for increased efficiency?

File: pytorch_lightning/loops/optimizer/optimizer_loop.py [hunk 2]
```diff
@@ -355,22 +330,3 @@ def _training_step(
...
-        should_track = float(self.trainer.track_grad_norm) > 0
```
Comment (carmocca) [inline pytorch_lightning/loops/optimizer/optimizer_loop.py:368]: Isn't `track_grad_norm == 0` a valid value?

File: pytorch_lightning/accelerators/accelerator.py [hunk 3]
```diff
@@ -304,17 +304,6 @@ def optimizer_zero_grad(self, current_epoch: int, batch_idx: int, optimizer: Opt
         model_ref = self.lightning_module
         model_ref.optimizer_zero_grad(current_epoch, batch_idx, optimizer, opt_idx)
 
-    def clip_gradients(
```
Comment (carmocca) [inline pytorch_lightning/accelerators/accelerator.py:307]: This would be unused now, called directly within the `precision_plugin`

File: tests/models/test_hooks.py [hunk 4]
```diff
@@ -554,7 +556,9 @@ def training_step(self, batch, batch_idx):
         dict(name="Callback.teardown", args=(trainer, model), kwargs=dict(stage="fit")),
         dict(name="teardown", kwargs=dict(stage="fit")),
     ]
-    assert called == expected
+    assert [c["name"] for c in called] == [e["name"] for e in expected]
+    for c, e in zip(called, expected):
+        assert c["name"] == e["name"]
```
Comment (carmocca) [inline tests/models/test_hooks.py:561]: FIXME: undo this change
